components/content-service/pkg/initializer/prebuild.go

```diff
@@ -80,61 +84,85 @@ func (p *PrebuildInitializer) Run(ctx context.Context, mappings []archive.IDMapp
 	src = csapi.WorkspaceInitFromPrebuild
 
 	// make sure we're on the correct branch
+
+	if p.Git != nil {
+		err = runGitInit(ctx, p.Git)
+	} else if p.Composite != nil {
+		for _, init := range p.Composite.Initializer {
+			if gitinit, ok := init.(*GitInitializer); ok {
+				anErr := runGitInit(ctx, gitinit)
+				if anErr != nil {
+					if err == nil {
+						err = anErr
+					} else {
+						log.Error("Git init for prebuild on "+gitinit.RemoteURI+" failed.", err)
+					}
+				}
+			}
+		}
```

Comment:
We can simplify the error handling here. Also, we should think about how we handle the case if the composite initializer is not what we expect it to be. Maybe, instead of using a composite initializer we should not generalise that much, and have a `repeated GitInitializer multi_repo` instead.

```suggestion
			gitinit, ok := init.(*GitInitializer)
			if !ok {
						// What do we do here? Return an error that this kind of setup isn't supported?
			}
			err := runGitInit(ctx, gitinit)
			if err != nil {
				break
			}
		}
```